File: src/HttPlaceholder.Domain/FullStubModel.cs
using YamlDotNet.Serialization;

namespace HttPlaceholder.Domain
{
    /// <summary>
    /// A class for storing a stub with its metadata.
    /// </summary>
    public class FullStubModel
    {
        /// <summary>
        /// Gets or sets the stub.
        /// </summary>
        [YamlMember(Alias = "stub")]
        public StubModel Stub { get; set; }

        /// <summary>
        /// Gets or sets the metadata.
        /// </summary>
        [YamlMember(Alias = "metadata")]
        // ReSharper disable once UnusedAutoPropertyAccessor.Global
        public StubMetadataModel Metadata { get; set; }
    }
}

File: gui2/src/domain/stub/full-stub-model.ts
import type { StubModel } from "@/domain/stub/stub-model";
import type { StubMetadataModel } from "@/domain/stub/stub-metadata-model";
import type { FullStubOverviewModel } from "@/domain/stub/full-stub-overview-model";

export interface FullStubModel extends FullStubOverviewModel {
  stub: StubModel;
  metadata: StubMetadataModel;
}
